src/psyclone/psyir/backend/sir.py

```diff
@@ -0,0 +1,408 @@
+        skip_nodes is set to False then raise an exception if an
+        unsupported node is found.
+
+        :param node: an unsupported PSyIR node.
+        :type node: subclass of :py:class:`psyclone.psyGen.Node`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        :raises VisitorError: if skip_nodes is set to False.
+
+        '''
+        if not self._skip_nodes:
+            raise VisitorError(
+                "Class SIRWriter method node_node(), unsupported node "
+                "found '{0}'".format(type(node)))
+        result = "{0}[ {1} start ]\n".format(self._nindent,
+                                             type(node).__name__)
+        self._depth += 1
+        for child in node.children:
+            result += self._visit(child)
+        self._depth -= 1
+        result += "{0}[ {1} end ]\n".format(self._nindent, type(node).__name__)
+        return result
+
+    def nemoloop_node(self, loop_node):
+        '''Supported NEMO loops are triply nested with particular indices (not
+        yet checked) and should contain a nemokern. If this is not the
+        case then it is not possible to translate so an exception is
+        raised.
+
+        :param loop_node: a nemoLoop PSyIR node.
+        :type loop_node: subclass of :py:class:`psyclone.nemo.NemoLoop`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        :raises VisitorError: if the loop is not triply nested with \
+        computation within the triply nested loop.
+
+        '''
+        # Check first loop has a single loop as a child.
+        loop_content = loop_node.loop_body.children
+        if not (len(loop_content) == 1 and
+                isinstance(loop_content[0], NemoLoop)):
+            raise VisitorError("Child of loop should be a single loop.")
+
+        # Check second loop has a single loop as a child.
+        loop_content = loop_content[0].loop_body.children
+        if not (len(loop_content) == 1 and
+                isinstance(loop_content[0], NemoLoop)):
+            raise VisitorError(
+                "Child of child of loop should be a single loop.")
+
+        # Check third loop has a single NemoKern as a child.
+        loop_content = loop_content[0].loop_body.children
+        if not (len(loop_content) == 1 and
+                isinstance(loop_content[0], NemoKern)):
+            raise VisitorError(
+                "Child of child of child of loop should be a NemoKern.")
+
+        # The interval values are hardcoded for the moment (see #470).
+        result = ("{0}interval = makeInterval(Interval.Start, Interval.End, "
+                  "0, 0)\n".format(self._nindent))
+        result += ("{0}bodyAST = makeAST([\n".format(self._nindent))
+        self._depth += 1
+        result += self.nemokern_node(loop_content[0])
+        self._depth -= 1
+        if result[-1] == "\n" and result[-2] == ",":
+            # Remove the additional comma as this is the last entry in
```

Comment:
Could you just do something like `result.rstrip(",\n")` here?

Reply:
Ooh, much nicer, thanks.

Reply:
It's like newly laid turf.